ccmbioinfo/lfs
# CARDS-2403: The yarn.lock file for aggregated-frontend should be checked into git

Without checking the `aggregated-frontend/src/main/frontend/yarn.lock` file into git, every time that someone builds a _fresh_ clone of CARDS, they cannot be sure of the exact versions of the Yarn packages that they will be using. This can result in unpredictable behavior including broken functionality resulting from breaking changes in upstream packages.

Testing Instructions
===============

1. Using a _fresh_ clone of this repository on the CARDS-2403 branch, build with `mvn clean install`. Verify that the build was sucessful.
2. Remove all files not tracked by git by running `git clean -dxf`.
3. Delete the `yarn.lock` file (`rm aggregated-frontend/src/main/frontend/yarn.lock`).
4. Run the build again with `mvn clean install`. This time the build should fail during the _Webpack_ step.
5. Run `git diff` and observe the cause of the build failure - the change in version from `8.10.0` to `8.11.1` for the `acorn` Yarn package.
